newrelic/common/utilization.py

```diff
@@ -265,21 +265,42 @@ def get_values(cls, response):
 class DockerUtilization(CommonUtilization):
     VENDOR_NAME = 'docker'
     EXPECTED_KEYS = ('id',)
-    METADATA_FILE = '/proc/self/cgroup'
-    DOCKER_RE = re.compile(r'([0-9a-f]{64,})')
+
+    METADATA_FILE_CGROUPS_V1 = '/proc/self/cgroup'
+    METADATA_RE_CGROUPS_V1 = re.compile(r'([0-9a-f]{64,})')
+    
+    METADATA_FILE_CGROUPS_V2 = '/proc/self/mountinfo'
+    METADATA_RE_CGROUPS_V2 = re.compile(r'^.*/docker/containers/([0-9a-f]{64,})/.*$')
 
     @classmethod
     def fetch(cls):
+        # Try to read from cgroups
         try:
-            with open(cls.METADATA_FILE, 'rb') as f:
+            with open(cls.METADATA_FILE_CGROUPS_V1, 'rb') as f:
                 for line in f:
                     stripped = line.decode('utf-8').strip()
                     cgroup = stripped.split(':')
                     if len(cgroup) != 3:
                         continue
                     subsystems = cgroup[1].split(',')
                     if 'cpu' in subsystems:
-                        return cgroup[2]
+                        contents = cgroup[2].split('/')[-1]
+                        match = cls.METADATA_RE_CGROUPS_V1.search(contents)
+                        if match:
+                            return match.group(1)
```

Comment:
It looks like this was grabbing `match.group(0)` before. Should this still be `match.group(0)`?

Reply:
Yeah those were equivalent but unnecessary, so I put it back to 0.